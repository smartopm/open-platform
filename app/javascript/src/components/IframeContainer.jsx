import React from 'react'
import PropTypes from 'prop-types'

export default function IframeContainer({ link, height, width }) {
  return (
    <div>
      <iframe
        data-testid="iframe"
        src={link} height={height} width={width}
        style={{ border: 0 }}
        />
    </div>
  )
}

IframeContainer.propTypes = {
  link: PropTypes.string.isRequired,
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired
}
