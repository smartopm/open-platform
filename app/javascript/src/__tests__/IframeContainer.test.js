import React from 'react'
import { mount } from 'enzyme'
import IframeContainer from '../components/IframeContainer'

describe('iframe container component', () => {
  const props = {
    link: 'https://app.doublegdp.com/',
    height: 200,
    width: 345
  }
  const iframe = mount(<IframeContainer {...props} />)
  it('renders correctly', () => {
    expect(iframe).toMatchSnapshot()
  })
  it('gets the correct props', () => {
    const { link, height, width } = iframe.props()
    expect(link).toBe(props.link)
    expect(height).toBe(props.height)
    expect(width).toBe(props.width)
  })
  it('renders only one div element and one iframe element ', () => {
    expect(iframe.find('iframe')).toHaveLength(1)
    expect(iframe.find('div')).toHaveLength(1)
  })
})
