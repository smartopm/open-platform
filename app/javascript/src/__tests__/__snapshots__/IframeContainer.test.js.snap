// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iframe container component renders correctly 1`] = `ReactWrapper {}`;
